From: =?utf-8?q?Bastien_Roucari=C3=A8s?= <rouca@debian.org>
Date: Sat, 21 Oct 2023 14:11:43 +0000
Subject: Add a debian policy

Install also other policy
---
 config/Makefile.am                            |  1 +
 config/{policy-open.xml => policy-debian.xml} | 24 ++++++++++++------------
 configure.ac                                  |  6 +++---
 3 files changed, 16 insertions(+), 15 deletions(-)
 copy config/{policy-open.xml => policy-debian.xml} (90%)

diff --git a/config/Makefile.am b/config/Makefile.am
index f6d116f..887fd94 100644
--- a/config/Makefile.am
+++ b/config/Makefile.am
@@ -62,6 +62,7 @@ CONFIG_EXTRA_DIST = \
   config/locale.xml \
   config/log.xml \
   config/mime.xml \
+  config/policy-debian.xml \
   config/policy-limited.xml \
   config/policy-open.xml \
   config/policy-secure.xml \
diff --git a/config/policy-open.xml b/config/policy-debian.xml
similarity index 90%
copy from config/policy-open.xml
copy to config/policy-debian.xml
index 22a0955..052561b 100644
--- a/config/policy-open.xml
+++ b/config/policy-debian.xml
@@ -88,8 +88,8 @@
   <policy domain="Undefined" rights="none"/>
   <!-- Set maximum parallel threads. -->
   <!-- <policy domain="resource" name="thread" value="2"/> -->
-  <!-- Set maximum time to live in seconds or mnemonics, e.g. "2 minutes". When
-       this limit is exceeded, an exception is thrown and processing stops. -->
+  <!-- Set maximum time in seconds. When this limit is exceeded, an exception
+       is thrown and processing stops. -->
   <!-- <policy domain="resource" name="time" value="120"/> -->
   <!-- Set maximum number of open pixel cache files. When this limit is
        exceeded, any subsequent pixels cached to disk are closed and reopened
@@ -98,27 +98,27 @@
   <!-- Set maximum amount of memory in bytes to allocate for the pixel cache
        from the heap. When this limit is exceeded, the image pixels are cached
        to memory-mapped disk. -->
-  <!-- <policy domain="resource" name="memory" value="256MiB"/> -->
+  <policy domain="resource" name="memory" value="1024MiB"/>
   <!-- Set maximum amount of memory map in bytes to allocate for the pixel
        cache. When this limit is exceeded, the image pixels are cached to
        disk. -->
-  <!-- <policy domain="resource" name="map" value="512MiB"/> -->
+  <policy domain="resource" name="map" value="2048MiB"/>
   <!-- Set the maximum width * height of an image that can reside in the pixel
        cache memory. Images that exceed the area limit are cached to disk. -->
-  <!-- <policy domain="resource" name="area" value="16KP"/> -->
+  <policy domain="resource" name="area" value="256MP"/>
   <!-- Set maximum amount of disk space in bytes permitted for use by the pixel
        cache. When this limit is exceeded, the pixel cache is not created
        and an exception is thrown. -->
-  <!-- <policy domain="resource" name="disk" value="1GiB"/> -->
+  <policy domain="resource" name="disk" value="2GiB"/>
   <!-- Set the maximum length of an image sequence.  When this limit is
        exceeded, an exception is thrown. -->
   <!-- <policy domain="resource" name="list-length" value="32"/> -->
   <!-- Set the maximum width of an image.  When this limit is exceeded, an
        exception is thrown. -->
-  <!-- <policy domain="resource" name="width" value="8KP"/> -->
+  <policy domain="resource" name="width" value="32KP"/>
   <!-- Set the maximum height of an image.  When this limit is exceeded, an
        exception is thrown. -->
-  <!-- <policy domain="resource" name="height" value="8KP"/> -->
+  <policy domain="resource" name="height" value="32KP"/>
   <!-- Periodically yield the CPU for at least the time specified in
        milliseconds. -->
   <!-- <policy domain="resource" name="throttle" value="2"/> -->
@@ -141,9 +141,9 @@
   <!-- don't read sensitive paths. -->
   <!-- <policy domain="path" rights="none" pattern="/etc/*"/> -->
   <!-- Indirect reads are not permitted. -->
-  <!-- <policy domain="path" rights="none" pattern="@*"/> -->
+  <policy domain="path" rights="none" pattern="@*"/>
   <!-- These image types are security risks on read, but write is fine -->
-  <!-- <policy domain="module" rights="write" pattern="{MSL,MVG,PS,SVG,TXT,URL,XPS}"/> -->
+  <!-- <policy domain="module" rights="write" pattern="{MSL,MVG,PS,SVG,URL,XPS}"/> -->
   <!-- This policy sets the number of times to replace content of certain
        memory buffers and temporary files before they are freed or deleted. -->
   <!-- <policy domain="system" name="shred" value="1"/> -->
@@ -153,6 +153,6 @@
   <!-- Set the maximum amount of memory in bytes that are permitted for
        allocation requests. -->
   <!-- <policy domain="system" name="max-memory-request" value="256MiB"/> -->
-  <!-- If the basename of path is a symbolic link, the open fails -->
-  <!-- <policy domain="system" name="symlink" rights="none" pattern="follow"/> -->
+  <policy domain="coder" rights="none" pattern="HTTP" />
+  <policy domain="coder" rights="none" pattern="HTTPS" />
 </policymap>
diff --git a/configure.ac b/configure.ac
index 813be4f..9942a18 100644
--- a/configure.ac
+++ b/configure.ac
@@ -837,10 +837,10 @@ m4_define([_MAGICK_SECURITY_POLICY],
 [m4_define([MAGICK_SECURITY_POLICY_DEFAULT], [m4_if($1, limited, limited, m4_if($1, secure, secure, open))])dnl
   AC_MSG_CHECKING([which variant of security policy to enforce])
   AC_ARG_WITH([security-policy],
-    [AS_HELP_STRING([--with-security-policy=open|limited|secure|websafe],
+    [AS_HELP_STRING([--with-security-policy=open|limited|secure|websafe|debian],
       [security policy variant to enforce, @<:@default=]MAGICK_SECURITY_POLICY_DEFAULT[@:>@.])],
     [case $withval in
-    open|limited|secure|websafe)
+    open|limited|secure|websafe|debian)
       security_policy=$withval
       ;;
     *)
@@ -854,7 +854,7 @@ m4_define([_MAGICK_SECURITY_POLICY],
   AC_MSG_RESULT([$security_policy])
 ]) # MAGICK_SECURITY_POLICY
 
-_MAGICK_SECURITY_POLICY([open])
+_MAGICK_SECURITY_POLICY([debian])
 AC_SUBST([MAGICK_SECURITY_POLICY])
 
 # Disable/Enable support for full delegate paths
